talib/_abstract.pxi

```diff
@@ -348,8 +357,31 @@ class Function(object):
         This is a shortcut to the outputs property that also allows setting
         the input_arrays dict and function parameters.
         """
+        # do not cache ta-func parameters passed to __call__
+        opt_inputs = deepcopy(self.__opt_inputs)
```

Comment:
Why do you ``deepcopy`` it?

And then after calling the function, it's caching ``self.__opt_inputs = opt_inputs``?

Reply:
(Concerned about the performance impact of duplicating any of the inputs, or are these just the floats/ints not the input_arrays)?

Reply:
Maybe I just misunderstood what you had in mind... what this is doing is remembering the values as they were before the call to `__call__`, and then restoring them afterwards (in case they got changed for *this* call)

`__opt_inputs` is a nested dict, so just a simple copy copies the nested dictionaries by reference, and therefore parameter values set later by `set_function_args` end up changing the copy too.

Reply:
That's correct, just the floats/ints for the optional function parameters (timeperiod and co). There's some metadata in there too (for each key in opt_inputs): 
```
{
        'name': name,
        'display_name': bytes2str(info.displayName),
        'type': info.type,
        'help': bytes2str(info.hint),
        'default_value': default_value,
        'value': None
    }
```
So I could update it to only remember the values, would probably be a bit more efficient.